src/request.js

```diff
@@ -120,12 +120,13 @@ function handleMessage(rawMessage, response) {
 		const req = observerObj.request;
 		response.body.__header = response.header;
 
-		if (response.header.statusCode !== 200) {
+		if (response instanceof Error) {
+			observer.onError(response);
+		} else if (response.header.statusCode !== 200) {
 			observer.onError({
 				err: response.body,
 				req,
 			});
-			observer.onCompleted(response.body);
```

Comment:
Calling onCompleted after calling onError does nothing. See the [observable contract](http://reactivex.io/documentation/contract.html) (specifically The Contract Governing Notifications) where it says

> An Observable may make zero or more OnNext notifications, each representing a single emitted item, and it may then follow those emission notifications by either an OnCompleted or an OnError notification, but not both. Upon issuing an OnCompleted or OnError notification, it may not thereafter issue any further notifications.

Also see https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/observer.md#rxobserverprototypechecked, which explains an opt-in way to verify that you're not doing anything that breaks the observable contract.